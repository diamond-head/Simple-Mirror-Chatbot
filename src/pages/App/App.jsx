import React from 'react';
import { ChatHeader, InputFooter, MessageList } from '../../components';

import BotProfilImage from '../../assets/images/profile-icon.jpg';
import { getItemFromLocalStorage, setItemToLocalStorage } from '../../utilities/utils';

const vertices = Object.freeze({
  source: "ME",
  target: "BOT"
});
const STORAGE_KEY = "messages";

export default function App() {
  const [messages, setMessages] = React.useState(getItemFromLocalStorage(STORAGE_KEY) || []);
  const [inputMessage, setInputMessage] = React.useState("");

  React.useEffect(() => {
    if (messages.length > 0) setItemToLocalStorage(STORAGE_KEY, messages);
  }, [messages, messages.length]);

  const handleInputValueChange = (e) => {
    setInputMessage(e);
  }

  const handleSendButtonClick = () => {
    const payload = {
      source: vertices.source,
      target: vertices.target,
      timestamp: new Date().getTime(),
      message: inputMessage
    };
    
    setInputMessage("");
    setMessages((prev) => ([...prev, payload]));
    setTimeout(() => {
      const botPayload = {
        source: vertices.target,
        target: vertices.source,
        timestamp: new Date().getTime(),
        message: inputMessage
      };
      setMessages((prev) => {
        const payload = [...prev, botPayload]
        return payload;
      });
    }, 300);
  }

  return (
    <div className="App__container">
      <div className="App__wrapper">
        <ChatHeader title="Amber" icon={BotProfilImage} />
        <MessageList
          messages={
            messages.sort((a, b) => a.timestamp - b.timestamp)
          }
          source={vertices.source} 
          target={vertices.target} 
        />
        <InputFooter
          onValueChange={handleInputValueChange}
          onSendButtonClick={handleSendButtonClick}
          defaultMessageValue={inputMessage}
        />
      </div>
    </div>
  );
};