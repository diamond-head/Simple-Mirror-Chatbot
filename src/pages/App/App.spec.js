/**
 * @jest-environment jsdom
*/

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import App from './App';

describe("App component", () => {
  it("app snapshot renders", () => {
    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders inner component", () => {
    const wrapper = mount(<App />);
    // expect(wrapper.find())
  });
});