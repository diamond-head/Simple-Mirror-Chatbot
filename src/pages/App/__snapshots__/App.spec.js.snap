// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component app snapshot renders 1`] = `
<div
  className="App__container"
>
  <div
    className="App__wrapper"
  >
    <div
      className="chat-header__container"
    >
      <div
        className="chat-header__icon"
      >
        <img
          alt="header-profile-icon"
          src="test-file-stub"
        />
      </div>
      <div
        className="chat-header__title"
      >
        Amber
      </div>
    </div>
    <div
      className="message-list__container"
    />
    <div
      className="input-footer__container"
    >
      <div
        className="controlled-input__container"
      >
        <textarea
          className="controlled-input__textarea"
          name="message-input-box"
          onChange={[Function]}
          placeholder="Type your message..."
          rows={1}
          value=""
        />
      </div>
      <img
        alt="send-button"
        className="send-button"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
